server/models/snippet.js

```diff
@@ -1,30 +1,62 @@
-const mongoose = require("mongoose");
+const mongoose = require('mongoose');
 
 const snippetSchema = new mongoose.Schema({
-	code: {
-		type: String,
-		required: true,
-		
-	},
-	author: {
-		type: mongoose.Schema.Types.ObjectId,
-		required: true,
-		ref: "User",
-	},
-	reviewer: {
-		type: mongoose.Schema.Types.ObjectId,
-		ref: "User",
-	},
-	title: {
-		type: String,
-		required: true,
-		default: "Unknown",
-	},
-	language: {
-		type: String,
-		required: true,
-	},
+  code: {
+    type: String,
+    required: true,
+  },
+  type: {
+    type: String,
+    required: true,
+    enum: ['request', 'receive'],
+  },
```

Comment:
Based on the discussion we had today, we don't need to have type field on the snippets. At the time of rendering in the front-end, we have access to the `user._id`. So we fetch all the snippets of which the user is `author`, and those would be the requested snippets. And in another call, we fetch all the snippets of which the user is `reviewer`, and those would be the responded snippets.

Reply:
I deleted it already